src/chttpd_db.erl

```diff
@@ -225,9 +225,23 @@ maybe_flush_changes_feed(Acc0, Data, Len) ->
     },
     {ok, Acc}.
 
-handle_compact_req(Req, _) ->
-    Msg = <<"Compaction must be triggered on a per-shard basis in CouchDB">>,
-    couch_httpd:send_error(Req, 403, forbidden, Msg).
+handle_compact_req(#httpd{method='POST'}=Req, Db) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    case Req#httpd.path_parts of
```

Comment:
this is the first time I've seen path matching within a function, everywhere else we do this at the top level of the function. So, please make this two separate clauses.